### PR TITLE
PBI 143 filter animation

### PR DESCRIPTION
**Pull request by adphillips2017**
Issue - #143 

Description of changes - 

@mentions of the person or team responsible for reviewing proposed changes.

TO-DO checklist before submitting your pull-request - 

- [ ] Are all acceptance criteria met ? 
- [ ] Has the code been documented properly ?
- [ ] Does component css-class structure conform to the Blue Dino style template i.e. top-level element .wrapper, .container and other inside css classes ?
- [ ] Have all checks passed ?
- [ ] Have you tested inter-browser functionality (should work on the latest Chrome, Firefox and Edge browsers) ?

### CODE REVIEW

File: src/app/modules/filters/filters-popover/filters-popover.component.ts [hunk 1]
```diff
@@ -1,38 +1,56 @@
...
 @Component({
   selector: 'ccf-filters-popover',
   templateUrl: './filters-popover.component.html',
   styleUrls: ['./filters-popover.component.scss']
 })
-export class FiltersPopoverComponent {
-  filtersVisible = false;
+export class FiltersPopoverComponent implements OnInit {
   filtersBoxVisible = false;
+  filterContainer:HTMLElement|null;
```
Comment (axdanbol) [inline src/app/modules/filters/filters-popover/filters-popover.component.ts]: Not angular way
```suggestion
  @ViewChild('filterContainer', { read: ElementRef }) filterContainer: ElementRef<HtmlElement>;
```

ElementRef has a member 'nativeElement' that is the HTMLElement

File: src/app/modules/filters/filters-popover/filters-popover.component.html [hunk 2]
```diff
@@ -7,8 +7,8 @@
       <mat-icon svgIcon="filter" class="icon"></mat-icon>
     </ng-template>
   </button>
-  <div class="popup-box slide-in" *ngIf="filtersBoxVisible">
-    <div class="popup-body fade-in" *ngIf="filtersVisible">
+  <div id="popupContainer">
```
Comment (axdanbol) [inline src/app/modules/filters/filters-popover/filters-popover.component.html]: Prefer classes over id
```suggestion
  <div class="popupContainer" #filterContainer>
```

(#filterContainer is for use with @ViewChild)

File: src/app/modules/filters/filters-popover/filters-popover.component.ts [hunk 3]
```diff
@@ -1,38 +1,56 @@
...
+  popupBody:HTMLElement|null;
+
+  ngOnInit() {
```
Comment (axdanbol) [inline src/app/modules/filters/filters-popover/filters-popover.component.ts]: Not needed with use of @ViewChild

File: src/app/modules/filters/filters-popover/filters-popover.component.ts [hunk 4]
```diff
@@ -1,38 +1,56 @@
...
-    return {
-      'filter-box': !this.filtersVisible,
-      'filter-box-expanded': this.filtersVisible
-    };
+  displayFilters(): void {
+    if (!this.filterContainer || !this.popupBody) {
+      return;
+    }
+
+    this.filterContainer.style.width = '35em';
```
Comment (axdanbol) [inline src/app/modules/filters/filters-popover/filters-popover.component.ts]: Why did you switch from using classes to setting the styles directly?

File: src/app/modules/filters/filters-popover/filters-popover.component.ts [hunk 5]
```diff
@@ -6,33 +6,40 @@ import { Component } from '@angular/core';
   styleUrls: ['./filters-popover.component.scss']
 })
 export class FiltersPopoverComponent {
-  filtersVisible = false;
   filtersBoxVisible = false;
+  popupContainerWidth = '0em';
```
Comment (bherr2) [inline src/app/modules/filters/filters-popover/filters-popover.component.ts]: Let's keep css stuff in css unless its truly needed. @axdanbol is there a better way to do this?

Comment (axdanbol) [reply src/app/modules/filters/filters-popover/filters-popover.component.ts]: On container add `[class.visible]="filtersBoxVisible"` then everything else should be possible to move to css.
I.e.
```scss
.popover-container {
  ...
  .popover-body {
     ...
  }

  &.visible {
    width: 35rem;

    .popover-body {
        opacity: 100%;
        ...
    }
  }
}

Comment (adphillips2017) [reply src/app/modules/filters/filters-popover/filters-popover.component.ts]: good idea.  I made those changes in the latest push.

**Pull request merged**